### DISCUSSION TITLE
Encrypting Multiple RDP messages in a Single SSL message

### DISCUSSION THREAD

**Discussion by AhmadSoboh** [FreeRDP/FreeRDP#10443] [Q&A]
Hi,
Is it ok (from Windows RDP server side) to have multiple RDP messages encrypted in the same SSL message?
I have implemented a RDP client, where it is possible to attach multiple RDP messages in one SSL message, but seems like Windows RDP server doesn't like it, is there any note regarding this in RDP spec?

**Comment by akallabeth** [comment]
@AhmadSoboh please explain what a `SSL message` is, your question kind of depends on what you understand under that term.

**Reply by AhmadSoboh** [reply]
@akallabeth in case we want SSL protocol to be used to encrypt the RDP session.
SSL message consists of SSL header and the encrypted data, how many RDP messages is allowed to be encrypted in the same SSL message?

**Comment by hardening** [comment]
@AhmadSoboh I guess you mean multiple PDUs in a single SSL record ? AFAICT that's not a problem and that must happen quite often with FreeRDP
